src/dfx/src/lib/models/canister.rs

```diff
@@ -432,12 +498,26 @@ fn build_canister_js(canister_id: &CanisterId, canister_info: &CanisterInfo) ->
                 &canister_info.get_name().to_uppercase(),
             );
 
-        match decode_path_to_str(&file.path()?)? {
+        match decode_path_to_str(&file.path().context("Path is not valid unicode.")?)
+            .context("Failed to decode path to str.")?
```

Comment:
would `.context("Failed to decode path to str.")` be better as a context directive on `decode_path_to_str`?